core/src/main/java/com/linecorp/armeria/server/logging/AccessLogComponent.java

```diff
@@ -56,7 +56,7 @@
  * An access log component to generate a log message.
  */
 @FunctionalInterface
-interface AccessLogComponent {
+public interface AccessLogComponent {
```

Comment:
The classes you made public are meant to be used internally. Please explain why you had to make them public.

Reply:
The things requested in #1631 was 1) a way to select client IP among proxied addresses and 2) modify existing `COMBINED` components to configure access log format.

>  I would've liked to just use the AccessLogFormats.COMBINED, make a copy of the list where I replaced the first entry, and then just create a method same as the AccessLogWriter.combined(), possibly via a
> ```java
>    static AccessLogWriter custom(List<AccessLogComponent> accessLogComponents) {
>        ImmutableList<AccessLogComponent> logComponents = ImmutableList.toImmutableList(accessLogComponents);
>        return requestLog -> AccessLogger.write(logComponents, requestLog);
>    }
> ```

Do you think we won't open the API?

Reply:
No. The reason why @tobias- has to do that was because we could not get the correct client address. Because we fixed `%a` in this pull request, there's nothing to do for access logs.

Reply:
Thanks for the reply! Let me rever this :-)

Reply:
fixed